fcrepo-configs/src/main/java/org/fcrepo/config/BasePropsConfig.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.fcrepo.config;
+
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.context.annotation.PropertySources;
+
+/**
+ * A base class for property configs
+ *
+ * @author dbernstein
+ * @since 6.0.0
+ */
+@PropertySources({
+        @PropertySource(value = FedoraPropsConfig.DEFAULT_FCREPO_CONFIG_FILE_DEFAULT_PROP_SOURCE,
+                ignoreResourceNotFound = true),
+        @PropertySource(value = FedoraPropsConfig.FCREPO_CONFIG_FILE_PROP_SOURCE, ignoreResourceNotFound = true)
+})
+abstract class BasePropsConfig {
+
+    public static final String DEFAULT_FCREPO_CONFIG_FILE_DEFAULT_PROP_SOURCE =
+            "file:fcrepo-home/config/fcrepo.properties";
```

Comment:
The idea here is that if the user fires up fedora without any configuration file,  fcrepo will create <cwd>/fcrepo-home.  Then the decide they want to override some props so they create  <cwd>/fcrepo-home/config/fcrepo.properties.
No need to change any command-line args.

If they decide they want fcrepo.home somewhere else they can use fcrepo.config.file and point to their new config file which contains a definition  fcrepo.home.   Even if they leave <cwd>/fcrepo-home/config/fcrepo.properties, any properties defined in the new prop file will override those in <cwd>/fcrepo-home/config/fcrepo.properties.

Reply:
I believe there is a problem of a circular dependency. We need to know where to look for the default file.  But if the location of the default file depends on the value of fcrepo.home which may be defined in the property file.

Reply:
When you say:
> The idea here is that if the user fires up fedora without any configuration file, fcrepo will create /fcrepo-home

..you are referring to "firing up" Fedora with either the one-click or jetty:run, no? In both cases, those Fedora's are strictly for short-term testing. Is there a need to have a "default configuration file location" in the context of short-term testing? Is it not just as easy to start Fedora in either of these ways with a system property that points to the desired config file?
```
MAVEN_OPTS="-Dfcrepo.config.file=/tmp/awoods/fcrepo.props"; mvn jetty:run -pl fcrepo-webapp/
```
or
```
java -Dfcrepo.config.file=/tmp/awoods/fcrepo.props -jar fcrepo-webapp/target/fcrepo-webapp-6.0.0-SNAPSHOT-jetty-console.jar --headless
```

I may be missing a use case, but it seems like it may be clearer to remove this default.

Reply:
I believe that @dbernstein added this on my request. I stand by it as a good idea, though I do agree that the Fedora home directory should not be hard coded. In my experience setting up applications, it is normal to override a default application home directory and have it resolve your data, config, etc locations relative to that. In my mind, this is a normal install experience and if you're going require setting one `-D` parameter then it should be `fcrepo.home` and not whatever parameter sets the location of a configuration file. In most cases, the configuration file should be in a known location relative to the application home.

Reply:
@pwinckles : I like the idea, it just seemed like the default config location should be relative to wherever the user set `fcrepo.home`... I believe is what you are also saying.
If the default config only supports jetty:run and the one-click, I am not sure I see the value.

Reply:
@pwinckles : Do you know of a way to get around the circular reference  issue? If so let's keep it in.  But if not, I'm thinking it may be better to remove the default properties file location (<ie <cwd>/fcrepo-home/config) and require the user to specify the location of the config file  using the fcrepo.config.file java option.

Reply:
@pwinckles :  actually - scratch that.  I think I have a good work around.